homeassistant/components/slack/__init__.py

```diff
@@ -40,7 +42,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     slack = WebClient(token=entry.data[CONF_API_KEY], run_async=True, session=session)
 
     try:
-        await slack.auth_test()
+        res = await slack.auth_test()
+        slack.url = res["url"]
```

Comment:
This appears to be using the slack client to store some integration state -- that is, these are not actually defined properties of the slack client that I can see, but just properties that are being set here to hold data.

These seem like they would be a fit for hass.data as data stored outside of the client?